src/Butler/Auth/OIDC.hs

```diff
@@ -146,12 +146,22 @@ loginServer os process sessions mkIndexHtml jwtSettings auth oidcenv mWorkspace
 
     setSession :: UserName -> SessionProvider -> Servant.Handler AuthResp
     setSession username provider = do
-        session <- liftIO $ runProcessIO os process $ newSession sessions provider username
-        liftIO (SAS.acceptLogin cookieSettings jwtSettings session.sessionID) >>= \case
-            Just r -> do
-                let page = workspaceUrl mWorkspace
-                pure $ r (script_ $ "window.location.href = " <> showT page)
-            Nothing -> error "oops?!"
+        foundSessions <- atomically $ lookupSessionByUser sessions localProvider "guest"
```

Comment:
Here it is problematic because there's not guaranty to get only one session. Also I don't know if the user changed its username.

I was planning to use lookupSession but I don't have a sessionID and I'm not sure I can easily get it during that flow as we are just after the IDP return to the sever.

There's still a way to pass json data to the IDP, the IDP will send it back to the server in oauthState (we do that in Monocle) and perhaps we could store the sessionID into that state ? However the "login" route handler should be able to receive the sessionID from the app. Then not sure If from the AppContext I could get the sessionID then pass it as a parameter to the login route ?

Reply:
or perhaps I missed something :/

Reply:
I see, then I guess we need to store the external id.

Reply:
Please have a look at #39. I propose to add the external username to the session provider so that we can reliably query if an external user already exists using the 'lookupByProvider' function, which I hope should work for this use-case.

I've also updated the code to ensure a local username is unique so that when creating a new session, if the requested username is already used, then a new username is generated (e.g. guest, guest_1, guest_2, ...). I think this should work better, let me know what you think.